src/MatchResult.php

```diff
@@ -114,16 +112,16 @@ public function groupOr(int $index, $default): string
     /**
      * Match group by name.
      *
-     * @param string $groupName
+     * @param int|string $groupName
      *
      * @return string
      *
      * @throws RegexFailed
      */
-    public function namedGroup(string $groupName): string
+    public function namedGroup($groupName): string
```

Comment:
`group` and `namedGroup` have the exact same implementation now, right? Can't we call `group` in this method to avoid duplication?

Reply:
I mean something like this:

```php
public function namedGroup($groupName): stringg
{
    return $this->group($groupName);
}
```

Instead of implementing the internals twice, and it wouldn't break BC.